### ISSUE TITLE
Improving Layout For App

### ISSUE THREAD

**Issue by ThePrimalPea** [Fox2Code/FoxMagiskModuleManager#39]
**MOVE THE SEARCH BUTTON**

![Move search button](https://user-images.githubusercontent.com/92656767/151338994-0bc96e26-8945-4379-bee3-3493cb65b1d7.png)

I suggest move the search button out of the app corner a little bit, it's will look better for rounded corner devices. Mostly all apps left the search button in the right side.

**EXTEND THE SEARCH BAR**

![Full bar](https://user-images.githubusercontent.com/92656767/151337334-4a8fbaa4-463b-408b-8d1e-bcdb0707e3ab.png)

The search bar should be longer to look fit with the screen

**Comment by Fox2Code** [comment]
While I'm a long time java developer and have experience with root, I'm not really good at making good UIs.

While I'm improving, this will probably not happen any time soon.

Same problem with the #34 request.

**Comment by Madis0** [comment]
I don't think it fits to the right because it would cover the toggle, delete and link buttons.
Would be better to just keep it in the top bar like Play Store or YouTube.

**Comment by samyak039** [comment]
would like to suggest some more things related to layout: **make the Download/Install button and Info button, bigger and separate them out further**.

It has happened with me a few times, I wanted to see the info but pressed the other button by mistake and boom, the module was installed.

**Comment by Fox2Code** [comment]
If someone want to do the changes, why not, but I'm not doing it.
I already struggle with the UI enough.

**Comment by Fox2Code** [comment]
Yeah I definitely need help.

**Comment by nift4** [comment]
Hey, I can offer some help in making the UI a bit more material design, if you want I can also move the UI to jetpack compose & kotlin, lmk if you are interested

**Comment by Fox2Code** [comment]
Telegram -> https://t.me/Fox2Code_Chat
I will definitely try to improve the app UI.

**Comment by Electric1447** [comment]
You should also add padding to the recyclerview items, and change the search icon to a solid color fab.

I will add a pull request when I have some time

**Comment by DerGoogler** [comment]
I have fixed the darkmode in #107

**Comment by Fox2Code** [comment]
There is no fix, there is only improvement. Nothing is perfect, but everything can be better.
Especially when talking about user interface because everyone have their personal preferences.

**Comment by nikolowry** [comment]
Ever since UI updates started to be pushed, I've been encountering an issue with the app bar (see screen recording): https://neeks.me/cloud/index.php/s/GxDdNgqCHntf9gy

Device: Pixel 5
OS: ProtonAOSP 12.3.1 (12L)

**Comment by Fox2Code** [comment]
@nikolowry I tried to fix it in https://github.com/Fox2Code/FoxMagiskModuleManager/commit/c3d8213ecf5d36b57d013cba8c6e36cb100f5516 but no guarantee it will fix the issue.

**Comment by nikolowry** [comment]
> @nikolowry I tried to fix it in [c3d8213](https://github.com/Fox2Code/FoxMagiskModuleManager/commit/c3d8213ecf5d36b57d013cba8c6e36cb100f5516) but no guarantee it will fix the issue.

Just updated to https://github.com/Fox2Code/FoxMagiskModuleManager/releases/tag/0.4.4, but unfortunately the issue was not fixed. Thanks for the speedy attempt though, feel free to @ me if you want me to check again

**Comment by nikolowry** [comment]
@Fox2Code I was able to pinpoint my issue, it's due to an atypical device configuration. I prefer to set my Pixel 5 "Font-Size" to "Small" and "Display" to "Default". Attaching a video demonstrate: https://neeks.me/cloud/index.php/s/z2sxTKXAfWrb5Zi

I don't usually target Android in my daily development life, but maybe Layout Inspector and Layout Validaton could help: https://developer.android.com/studio/debug/layout-inspector

**Comment by Fox2Code** [comment]
Making layout adapt to every devices take time, I just put assumptions on layout  to not take too long to release the next update, I prefer making releases faster, even if it contains some bugs.

**Comment by nikolowry** [comment]
@Fox2Code there are thousands of different device configurations, so I was not suggesting to target my specific configuration -- but rather figure out the "right way" to calculate UI dimensions:

- https://blog.prototypr.io/designing-for-multiple-screen-densities-on-android-5fba8afe7ead
- https://medium.com/programming-lite/best-practices-for-supporting-android-application-in-multiple-screen-a685afa83493
- https://blog.mindorks.com/understanding-density-independent-pixel-sp-dp-dip-in-android

**Comment by nikolowry** [comment]
Happy to report the DPI issues I was experiencing has been resolved in one of the latest releases. Thanks!
